### PR TITLE
[Enhancement] Support microsecond in DateLiteral

### PR DESCRIPTION
**Pull request by alvin-coding**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

I supported microsecond in Literal in this PR. 

After applying this CL, the following statement can ingest timestamp data with microseconds.
```
insert into test_ts values (1, "2021-01-01 12:34:45.123456");
```

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/analysis/DateLiteral.java [hunk 1]
```diff
@@ -344,7 +368,6 @@ public Expr uncheckedCastTo(Type targetType) throws AnalysisException {
         } else if (Type.isImplicitlyCastable(this.type, targetType, true)) {
             return new CastExpr(targetType, this);
         }
-        Preconditions.checkState(false);
```
Comment (silverbullet233) [inline fe/fe-core/src/main/java/com/starrocks/analysis/DateLiteral.java:347]: why remove this line?

Comment (alvin-coding) [reply fe/fe-core/src/main/java/com/starrocks/analysis/DateLiteral.java]: Changed

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 50 / 59 (84.75%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/Explain.java|0|7|00.00%|[733, 734, 735, 736, 738, 739, 740]|
|:large_blue_circle:|com/starrocks/analysis/DateLiteral.java|47|49|95.92%|[216, 435]|
|:large_blue_circle:|com/starrocks/catalog/PartitionKey.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/ColumnFilterConverter.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/plan/ScalarOperatorToExpr.java|1|1|100.00%|[]|

**Comment by silverbullet233** [comment]
run starrocks_admit_test

**Review by kangkaisen — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Pull request merged**